### PR TITLE
Commit engine.realm as empty

### PR DESCRIPTION
**Pull request by jtmcdole**
- only set it when the environment FLUTTER_REALM is present

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,7 +3,6 @@
 
 # This is dynamic in a monorepo
 bin/internal/engine.version
-bin/internal/engine.realm
```
Comment (christopherfujino) [inline .gitignore:6]: Would this leave user's checkouts dirty?

File: bin/internal/update_dart_sdk.sh [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #
 # -------------------------------------------------------------------------- #
 
-set -e
+set -ex
```
Comment (christopherfujino) [inline bin/internal/update_dart_sdk.sh:15]: did you mean to leave the `-x` here? I think this would break things.

File: .gitignore (hunk 1 above)
Comment (jtmcdole) [reply .gitignore:6]: Only if the user has FLUTTER_REALM set.

File: bin/internal/update_dart_sdk.sh (hunk 2 above)
Comment (jtmcdole) [reply bin/internal/update_dart_sdk.sh:15]: I did not.

File: .gitignore (hunk 1 above)
Comment (christopherfujino) [reply .gitignore:6]: oh wait, nvm, i guess we'd only write this on CI?

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by yjbanov — APPROVED** [review]
lgtm

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**